client/my-sites/site-settings/form-general.jsx

```diff
@@ -113,57 +117,135 @@ const FormGeneral = React.createClass( {
 			amp_is_supported: false,
 			amp_is_enabled: false,
 		} );
-	},
+		this.props.clearDirtyFields();
+		this.props.updateFields( this.getFormSettings( this.props.settings ) );
+	}
 
-	onTimezoneSelect( timezone ) {
-		this.setState( { timezone_string: timezone } );
-	},
+	componentWillReceiveProps( nextProps ) {
+		this._showWarning( nextProps.site );
+
+		if ( nextProps.settings !== this.props.settings ) {
+			let newState = this.getFormSettings( nextProps.settings );
+			//If we have any fields that the user has updated,
+			//do not wipe out those fields from the poll update.
+			newState = omit( newState, nextProps.dirtyFields );
+			nextProps.updateFields( newState );
+		}
+
+		if (
+			this.props.isSavingSettings &&
+			! nextProps.isSavingSettings &&
+			nextProps.saveRequestStatus === 'success' ) {
```

Comment:
Could have the component be unaware of the specific status (especially since not constant) by further abstracting a `isSiteSettingsSaveSuccessful` selector. Might be a little too far abstracted 😄